packages/next-on-pages/src/buildApplication/generateFunctionsMap.ts

```diff
@@ -703,3 +809,26 @@ async function copyWasmFiles(
 		await copyFile(originalFileLocation, newLocation);
 	}
 }
+
+/**
+ * Creates js files into the __next-on-pages-dist__/nextjs-manifest folder which export a default
+ * value containing a Next.js manifest object
+ *
+ * @param distDir the __next-on-pages-dist__ directory's path
+ * @param manifests map mapping next manifest identifiers to the manifest objects
+ */
+async function createNextJsManifestFiles(
+	distDir: string,
+	manifests: Map<string, string>
+): Promise<void> {
+	const manifestsDistDir = join(distDir, 'nextjs-manifests');
+	// TODO: the following `mkdir` should not be recursive (since at this point the distDir should always exist),
+	//       the current unit tests force us to make this recursive, when we get proper e2es we need to phase out
+	//       the use of mocking fs in our unit tests and this should hopefully allow us to remove unnecessary
+	//       "recursive" flags in `mkdir` calls such as the following
```

Comment:
I put this as I thought of giving up on removing the `recursive`... I need to give it another shot before throwing the towel 😓

Reply:
Update:

The main issue here is the test with the custom dir, it seems like it doesn't create the correct directory, and if I try to add the directory myself via mockFs it gets wiped here:
https://github.com/cloudflare/next-on-pages/blob/68d905ed16c92aa36ad4b376639eebcd969cd267/packages/next-on-pages/src/buildApplication/processVercelOutput.ts#L88-L94

ok so the only possible solution is to get the test to trigger the prod code to generate the correct directory... can we do it... 🤔

Reply:
ok it was actually my bad 😅 

it seems like we might not generate the output directory (`__next-on-pages-dist__`) if we only have static/prerendered routes, which is what that test has

so instead I added a check to only create the manifest dir if we've actually found manifests to create (which I thought we'd always do, but on hindsight yeah if everything is static I guess no manifests will be needed 🤔)